src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class TransportWithMessageCredentialNetTCPTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public TransportWithMessageCredentialNetTCPTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact(Skip = "Implement certificate generation and enable for both platforms")]
+        public void BasicUserNameAuth()
```

Comment:
There's already a test service certifcate that's generated by the dotnet sdk and is available. I'm not sure how to fetch it, e.g. is there a naming convention for example, but I know Kestrel is well capable of getting it. Maybe add a breakpoint on SslStream.AuthenticateAsServer for an HTTPS test and then work backwards to find the code where it's retrieved. Or a better breakpoint might be in the X509Certificate2 constructor. You don't need a client certificate for this scenario.